lib/icedfrisby.js

```diff
@@ -1271,11 +1316,9 @@ class Frisby {
       try {
         this.current.expects[i].call(null)
       } catch (e) {
-        if (false === this._exceptionHandler) {
-          this.current.failures.push(e)
+        const shouldStop = this._noteExpectationFailed(e)
+        if (shouldStop) {
           break
-        } else {
-          this._exceptionHandler.call(this, e)
```

Comment:
Factored this out rather than duplicating it. New code is equivalent to old code.